filter/filter.go

```diff
@@ -31,13 +35,55 @@ func (f *Filter) IsExcluded(v string) bool {
 	return ok
 }
 
+// Validate validates that the data inside of the filters is right
+func (f *Filter) Validate() error {
+	// Validate that the Targets have the right format
+	if len(f.Targets) != 0 {
+		for _, t := range f.Targets {
+			split := strings.Split(t, ".")
+			// IDs can have . in between so we validate that at least we have the minimum
+			if len(split) < 2 {
+				return errors.Wrapf(errcode.ErrFilterTargetsInvalid, "the Target %q has an invalid format. The expected format is 'aws_instance.ID'", t)
+			}
+		}
+	}
+
+	return nil
+}
+
+// TargetsTypesWithIDs returns all the types (ex: aws_instance) from
+// the list of Targets and the IDs
+func (f *Filter) TargetsTypesWithIDs() map[string][]string {
+	types := make(map[string]map[string]struct{})
+	res := make(map[string][]string)
+
+	for _, t := range f.Targets {
+		split := strings.Split(t, ".")
+		ty := split[0]
+		id := strings.Join(split[1:], ".")
```

Comment:
I guess this could be explained, e.g. AWS IAM users with `firstname.lastname`
The split could also be replaced by `Split` variations to avoid making a join afterwards, like: `strings.SplitN("aws_iam_user.firstname.lastname", ".", 2)` which would do the same operation I believe.

Reply:
I did not remember of the `SplitN` much better yep!.

Reply:
Yes, I feel like it make sense to have it here